Create 2025-04-25-solidity-developer-survey-2024-results.md (#141)
ethereum/solidity-website: src/posts/2025-04-25-solidity-developer-survey-2024-results.md

In total, 684 participants from 90 different countries participated in the 2024 survey. 

The coverage of different geographies increased from 73 countries in 2021, to 100 countries in 2022, and then back to 71 in 2023, it has now increased to 90 countries this year. It is interesting to note a spike in the geographic diversity of the survey audience this year compared to 2023.

### Residency

A majority of 10.8% of respondents mentioned that they live in the USA, followed by India and Nigeria at 7.8% each, and France taking over Germany this year with 3.8% votes.

Except for France ranking higher than Germany this year, the order of the top 5 countries of residence remains similar to 2023. However, there has been a shift in individual numbers. For instance, there's a drop in USA residents among the survey participants, whereas Nigeria has seen a spike.

For reference, see the world map and chart with a list of countries of residence with 13+ responses.

![Participants World Map](/img/2025/survey/Slide3.png)

![List of Countries with 13+ responses](/img/2025/survey/Slide4.png)

> 
> 

Author: ```suggestion

```

- This is our fifth annual survey. You can compare the 2024 results with the previous years by checking our blog posts below:
    - [2023](https://soliditylang.org/blog/2024/04/03/solidity-developer-survey-2023-results/)
    - [2022](https://soliditylang.org/blog/2023/03/10/solidity-developer-survey-2022-results/)
    - [2021](https://blog.soliditylang.org/2022/02/07/solidity-developer-survey-2021-results/)
    - [2020](https://blog.soliditylang.org/2021/01/26/solidity-developer-survey-2020-results/)

Without further ado, let’s dig into the 2024 results!

## Summary & Notable Insights

The survey consists of 6 sections, namely: **[1]** Demographics, **[2]** Dev & User Profiles, **[3]** Solidity Background, **[4]** Solidity Developer Experience, **[5]** Language Design, and **[6]** Solidity Developer Community.

Let's look at some key insights from the summary of these various sections from the 2024 survey.

> - **Demographics**: In total, 684 developers from 91 different countries participated in the 2023 survey. The highest number of respondents hail from the **USA (10.8%)**, followed by **India and Nigeria at 7.8% each**. This remains consistent with last year's insights, except for Nigeria sharing the second place alongside India this year.

Author: ```suggestion
- **Demographics**: In total, 684 developers from 91 different countries participated in the 2024 survey. The highest number of respondents hail from the **USA (10.8%)**, followed by **India and Nigeria at 7.8% each**. This remains consistent with last year's insights, except for Nigeria sharing the second place alongside India this year.
```

In total, 684 participants from 90 different countries participated in the 2024 survey. 

The coverage of different geographies increased from 73 countries in 2021, to 100 countries in 2022, and then back to 71 in 2023, it has now increased to 90 countries this year. It is interesting to note a spike in the geographic diversity of the survey audience this year compared to 2023.

### Residency

A majority of 10.8% of respondents mentioned that they live in the USA, followed by India and Nigeria at 7.8% each, and France taking over Germany this year with 3.8% votes.

Except for France ranking higher than Germany this year, the order of the top 5 countries of residence remains similar to 2023. However, there has been a shift in individual numbers. For instance, there's a drop in USA residents among the survey participants, whereas Nigeria has seen a spike.

For reference, see the world map and chart with a list of countries of residence with 13+ responses.

![Participants World Map](/img/2025/survey/Slide3.png)

![List of Countries with 13+ responses](/img/2025/survey/Slide4.png)

> Alt-text: List of countries of residence with 13+ responses

Author: ```suggestion

```

A total of 25.8% of users contribute to open source projects written in Solidity regularly (daily and weekly), whereas 28.4% only contribute monthly.

On the other hand, 45.7% of users reported that they have never contributed to open-source projects written in Solidity.

[Compared to previous years](https://soliditylang.org/img/2024/survey/Slide17.png), we are unfortunately seeing a downtrend of users who contribute to open-source projects written in Solidity.

![Open Source Contributions](/img/2025/survey/Slide17.png)

### Programming Language Preferences

Like previous years, Solidity ranks as the most used programming language among the participants (42.4%), [a number almost identical to last year's](https://soliditylang.org/img/2024/survey/Slide18.png), followed by TypeScript (21.5%), and JavaScript (15%).

The order of the top 3 most used languages remains the same as last year.

Other less frequently mentioned languages are Python (10.3%), Rust (4.4%), Go (2%), C# (1.7%), and Java (1.7%).

Since Go was specified as an "Other" response, almost as many times as Java last year, it was introduced as a choice in the list of languages and ranked higher than C# and Java.

> Please note that there are missing labels of other languages that did not show up in the chart due to space constraints. You can find the [full list in the analysis sheet](link).

Author: ```suggestion
Please note that there are missing labels of other languages that did not show up in the chart due to space constraints. You can find the [full list in the analysis sheet](https://docs.google.com/spreadsheets/d/1ptivw3pC-IMJ9Bhl6lViyKd-NTEDZGhE3-bwBFtMwtE/edit?usp=sharing).
```

![Other Smart Contract Languages](/img/2025/survey/Slide53.png)

![header 4](/img/2024/survey/header4.png)

## Solidity Developer Experience

In this section of the survey, we will be going through the developer experience of our user community.

We asked about the overall improvement in the Solidity developer experience.

66.8% of all respondents generally believe that it has improved in the last year, with 24.2% even believing it was a major improvement.

Only 1.5% of the respondents believe that the developer experience has gotten worse.

![Solidity Developer Experience](/img/2025/survey/Slide55.png)

### Recurring Issues

When asked if the participants encounter recurring issues, 60.7% of the total respondents said that they do not come across the same or similar issues multiple times when developing in Solidity.

Among the 39.3% that encounter recurring issues, Stack too deep issues are encountered most frequently, followed by debugging issues, bytecode size limit, and optimiser-related issues.

![Recurring issues](/img/2025/survey/Slide56.png)

![Recurring issues breakdown](/img/2025/survey/Slide57.png)

> _ℹ️ **Note:** On the topic of debugging issues, we would like to remind you about the ethdebug standards development working group, an initiative aimed at defining a common debugging data format for languages built on top of the EVM: [ethdebug](https://github.com/ethdebug/format/).

Author: It is the Italics formatting

> _ℹ️ **Note:** On the topic of debugging issues, we would like to remind you about the ethdebug standards development working group, an initiative aimed at defining a common debugging data format for languages built on top of the EVM: [ethdebug](https://github.com/ethdebug/format/).

Reviewer: doesn't come out that way on the blog

> _ℹ️ **Note:** On the topic of debugging issues, we would like to remind you about the ethdebug standards development working group, an initiative aimed at defining a common debugging data format for languages built on top of the EVM: [ethdebug](https://github.com/ethdebug/format/).

Reviewer: it has to be paragraph-wise, so

```suggestion
_ℹ️ **Note:** On the topic of debugging issues, we would like to remind you about the ethdebug standards development working group, an initiative aimed at defining a common debugging data format for languages built on top of the EVM: [ethdebug](https://github.com/ethdebug/format/)._
```

same for the paragraph below

> _ℹ️ **Note:** On the topic of debugging issues, we would like to remind you about the ethdebug standards development working group, an initiative aimed at defining a common debugging data format for languages built on top of the EVM: [ethdebug](https://github.com/ethdebug/format/).

Author: Weird. I'll remove it.
